### PR TITLE
MTM-43486 Review Data explorer

### PR DESCRIPTION
**Pull request by Airdroplet**
* improved description in some places
* updated screenshots
* removed some unclear screenshots

### CODE REVIEW

**Comment by Airdroplet** [comment]
@BeateRixen Review of this part is quite complex. I suspect that there are some issues for UI to fix in order to make sure that Data Explorer, at least in relation to information provided within internal doc: https://docs.google.com/document/d/1rWydw6kHDJnzVk12-NClFT4JVAjfMJpTVaxBdK9w_ng/edit?usp=sharing

Therefor please review carefully changes suggested here.

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 1]
```diff
@@ -46,6 +46,8 @@ There can be more than one matching data point entry in the data point library.
 
 For details on modifying the visualization in general, see [Changing data explorer visualization](#change-visualization). For details on customizing the properties of a particular data point, see [Customizing data point properties](#customize-data-points).
 
+>**Info:** Save as default is triggering save of datapoint configuration within particular source device object. In case of saving configuration as default, reference to stored data can be used by other parts of system, for example by smart rules. Note also that in order to create threshold smartrule you need to save default at least once for selected data point.
```
Comment (BeateRixen) [inline content/users-guide/cockpit-bundle/data-explorer.md:49]: @Airdroplet Where do I find this "Save as default" option?

**Comment by Airdroplet** [comment]
@BeateRixen please don't rush with merging this one. I still need to talk with team in respect to some issues observed on platform in respect to this feature. Hopefully will finish with my tests before Friday.

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 2]
```diff
@@ -40,12 +40,17 @@ The data points properties are pre-filled as follows:
...
 * If the data points have a matching definition in the data point library, the values from the data point library are used.
 
-There can be more than one matching data point entry in the data point library. In this case, the first one is selected automatically by the system. You can overwrite this selection by clicking the menu icon of the respective data point and selecting **Load [NAME] from Library**.
+There can be more than one matching data point entry in the data point library. In this case, the first one is selected automatically by the system. You can overwrite this selection by clicking the menu icon of the respective data point and selecting **Load [NAME] from library**.
 
 ![edit data points](/images/users-guide/cockpit/cockpit-dataexplorer-edit.png)
 
 For details on modifying the visualization in general, see [Changing data explorer visualization](#change-visualization). For details on customizing the properties of a particular data point, see [Customizing data point properties](#customize-data-points).
 
+Click **Save as default** if you want for each user accessing the view to be able to see data points configuration.
```
Comment (Airdroplet) [inline content/users-guide/cockpit-bundle/data-explorer.md:49]: @BeateRixen I have changed this part to correspond to previously mentioned internal document. 

Additionally did some testing due to previous doubts and confirm that everything is fine there.

**Comment by Airdroplet** [comment]
@BeateRixen I think it is fine now from content perspective. Please verify grammar and it can go in :)

File: content/users-guide/cockpit-bundle/data-explorer.md (hunk 2 above)
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md:49]: hm, still not right :)

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 3]
```diff
@@ -40,12 +40,17 @@ The data points properties are pre-filled as follows:
...
 
-There can be more than one matching data point entry in the data point library. In this case, the first one is selected automatically by the system. You can overwrite this selection by clicking the menu icon of the respective data point and selecting **Load [NAME] from Library**.
+There can be more than one matching data point entry in the data point library. In this case, the first one is selected automatically by the system. You can overwrite this selection by clicking the menu icon of the respective data point and selecting **Load [NAME] from library**.
 
 ![edit data points](/images/users-guide/cockpit/cockpit-dataexplorer-edit.png)
 
 For details on modifying the visualization in general, see [Changing data explorer visualization](#change-visualization). For details on customizing the properties of a particular data point, see [Customizing data point properties](#customize-data-points).
 
+Click **Save as default** if you want each user accessing the view to be able to see predefined configuration when adding datapoints to be displayed in data explorer.
```
Comment (Airdroplet) [inline content/users-guide/cockpit-bundle/data-explorer.md:49]: @BeateRixen this has been changed

File: content/users-guide/cockpit-bundle/data-explorer.md (hunk 2 above)
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md]: changed once more

**Comment by Airdroplet** [comment]
@BeateRixen now it should be correct :)

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 4]
```diff
@@ -102,15 +107,13 @@ In the top of the dialog, select a device from the asset hierarchy. Only the ass
 
 The bottom of the dialog shows all data points of the selected object. Select the data points you want to show in the data explorer. Click **Add** to add all selected data points to the list of data points.
 
-JsonPath expressions added from data points will be stored in bracket-notation in order to increase flexibility in fragment and series naming (e.g. whitespaces will be supported):
-
-![Measurement added from data point](/images/users-guide/cockpit/cockpit-dataexplorer-adddatapoint-measurement.png)
+>**Info:** JsonPath expressions added from data points are stored in bracket notation in order to increase flexibility in fragment and series naming (for example whitespaces are supported):
```
Comment (pawel-rynarzewski-sag) [inline content/users-guide/cockpit-bundle/data-explorer.md:110]: The trailing `:` suggests that something will follow it, e.g. a screenshot, but now it's removed. I see the screenshot was from export UI, so not really belonging in data point explorer section. Shouldn't we move both JsonPath note and the screenshot to the right section instead?

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 5]
```diff
@@ -40,12 +40,17 @@ The data points properties are pre-filled as follows:
...
 
 ![edit data points](/images/users-guide/cockpit/cockpit-dataexplorer-edit.png)
 
 For details on modifying the visualization in general, see [Changing data explorer visualization](#change-visualization). For details on customizing the properties of a particular data point, see [Customizing data point properties](#customize-data-points).
 
+Click **Save as default** if you want each user accessing the view to be able to see predefined configuration when adding datapoints to be displayed in data explorer.
+
+>**Info:** if you have existing **on measurement threshold create alarm** smartrule based on data point library entry used for current datapoint, when you click **save as default** you are effectively overriding configuration from library with current values provided for that particular datapoint. It means that on clicking **save as default** in case of smartrule existing, conditions for creating alarm are taken from data point configuration of current device instead of data point library.
```
Comment (pawel-rynarzewski-sag) [inline content/users-guide/cockpit-bundle/data-explorer.md:51]: Maybe change "Info" to "Example"?

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 6]
```diff
@@ -40,12 +40,17 @@ The data points properties are pre-filled as follows:
...
+If you want to restore data point library default settings, click **Load from library** and then click **Save as default**.
```
Comment (pawel-rynarzewski-sag) [inline content/users-guide/cockpit-bundle/data-explorer.md:52]: I'd move this to the main text above.

File: content/users-guide/cockpit-bundle/data-explorer.md (hunk 3 above)
Comment (pawel-rynarzewski-sag) [inline content/users-guide/cockpit-bundle/data-explorer.md:49]: This is only one of the effects. Maybe: `Click **Save as default** to store data point settings into device's managed object. These settings will then take precendence over the settings from data point library, e.g. when this data point is added to a data explorer by other users, or when there is a threshold smart rule processing this device and its data point. If you want to restore data point library default settings, click **Load from library** and then click **Save as default**.`?
btw, executing the last sentence, only save settings from library to device, right? There is no way to genuinely remove data point entry from device's managed object?

**Review by pawel-rynarzewski-sag — CHANGES_REQUESTED** [review]
General remark: please use "data point" and "smart rule" consistently with the rest of documentation.

File: content/users-guide/cockpit-bundle/data-explorer.md (hunk 3 above)
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md:49]: only manually via rest

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 7]
```diff
@@ -102,15 +104,13 @@ In the top of the dialog, select a device from the asset hierarchy. Only the ass
...
 The bottom of the dialog shows all data points of the selected object. Select the data points you want to show in the data explorer. Click **Add** to add all selected data points to the list of data points.
 
-JsonPath expressions added from data points will be stored in bracket-notation in order to increase flexibility in fragment and series naming (e.g. whitespaces will be supported):
-
-![Measurement added from data point](/images/users-guide/cockpit/cockpit-dataexplorer-adddatapoint-measurement.png)
+>**Info:** JsonPath expressions added from data points are stored in bracket notation in order to increase flexibility in fragment and series naming (for example whitespaces are supported):
 
 To save the data point to the data point library, click the menu icon of the data point and from the context menu select **Save to library**.
 
 ![Save data point to library](/images/users-guide/cockpit/cockpit-dataexplorer-savedatapoint.png)
```
[the remark this answers is not in the document]
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md:114]: yeah, right, will replace that one.

File: content/users-guide/cockpit-bundle/data-explorer.md (hunk 5 above)
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md:51]: done

File: content/users-guide/cockpit-bundle/data-explorer.md (hunk 4 above)
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md:109]: removed from here

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 8]
```diff
@@ -102,15 +106,11 @@ In the top of the dialog, select a device from the asset hierarchy. Only the ass
...
-JsonPath expressions added from data points will be stored in bracket-notation in order to increase flexibility in fragment and series naming (e.g. whitespaces will be supported):
-
-![Measurement added from data point](/images/users-guide/cockpit/cockpit-dataexplorer-adddatapoint-measurement.png)
-
 To save the data point to the data point library, click the menu icon of the data point and from the context menu select **Save to library**.
 
 ![Save data point to library](/images/users-guide/cockpit/cockpit-dataexplorer-savedatapoint.png)
 
-For details on the data point library refer to [Using the Datapoint Library](#data-point-library).
+For details on the data point library refer to [Using the datapoint library](#data-point-library).
```
Comment (pawel-rynarzewski-sag) [inline content/users-guide/cockpit-bundle/data-explorer.md:113]: "Data point library" - this is how it's spelled in the app

File: content/users-guide/cockpit-bundle/data-explorer.md [hunk 9]
```diff
@@ -40,12 +40,16 @@ The data points properties are pre-filled as follows:
...
 
+Click **Save as default** to store data point settings into device's managed object. These settings will then take precendence over the settings from data point library, e.g. when this data point is added to a data explorer by other users, or when there is a threshold smart rule processing this device and its data point. If you want to restore data point library default settings, click **Load from library** and then click **Save as default**.
+
+>**Example:** if you have existing **on measurement threshold create alarm** smartrule based on data point library entry used for current datapoint, when you click **save as default** you are effectively overriding configuration from library with current values provided for that particular datapoint. It means that on clicking **save as default** in case of smartrule existing, conditions for creating alarm are taken from data point configuration of current device instead of data point library.
```
[the remark this answers is not in the document]
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md:51]: Sounds good.

File: content/users-guide/cockpit-bundle/data-explorer.md (hunk 8 above)
Comment (Airdroplet) [reply content/users-guide/cockpit-bundle/data-explorer.md:113]: changed

**Review by pawel-rynarzewski-sag — APPROVED** [review]

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**